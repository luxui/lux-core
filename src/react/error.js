/**
 * @module react/error
 * @memberof react
 */

import React from 'react'; // `React` must be in scope when using JSX

import registry from '../lib/componentRegistry';

function errorComponent(props) {
  // eslint-disable-next-line no-console
  console.log(props);

  return (
    <section>
      <h1>Error</h1>
      <p>Luxui encountered an error.</p>
      <p><strong>{`${props.error}`}</strong></p>
    </section>
  );
}
errorComponent.propTypes = {
  error: React.PropTypes.oneOfType([
    React.PropTypes.string,
    React.PropTypes.instanceOf(Error),
  ]),
};

registry('Lux.Error', errorComponent, false);
