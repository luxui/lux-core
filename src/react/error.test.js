import React from 'react';
import renderer from 'react-test-renderer';

import registry from '../lib/componentRegistry';

import './error';
const Component = registry('Lux.Error');

describe('Error - main - Component', function () {
  it('should exist; and should be a function', function () {
    expect(typeof Component).toBe('function');
  });

  it('should match the snapshot', function () {
    spyOn(console, 'log');

    const response = {
      data: {},
      error: new Error('Unforseen error.'),
      path: '/',
      status: 400,
    };

    const component = renderer.create(
      <Component {...response} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
