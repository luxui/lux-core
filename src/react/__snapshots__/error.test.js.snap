// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error - main - Component should match the snapshot 1`] = `
<section>
  <h1>
    Error
  </h1>
  <p>
    Luxui encountered an error.
  </p>
  <p>
    <strong>
      Error: Unforseen error.
    </strong>
  </p>
</section>
`;
